cf/aggregate.py

```diff
@@ -3422,6 +4121,46 @@ def _ok_coordinate_arrays(
     return True
 
 
+def _difference_units(units):
+    """Return difference units corresponding to position-on-scale units.
+
+    .. versionadded:: TODOAGGVER
+
+    :Parameters:
+
+        units: `Units`
+            The position-on-scale units.
+
+    :Returns:
+
+        `Units`
+            The difference units.
+
+    **Examples**
+
+    >>> _difference_units('km')
+    <Units: km>
+    >>> _difference_units('day')
+    <Units: day>
+    >>> _difference_units('K')
+    <Units: K>
+    >>> _difference_units('degC')
+    <Units: degC>
+
+    >>> _difference_units('hours since 2000-01-01')
+    <Units: hours>
+
+    """
+    if units.isreftime:
+        return Units(units._units_since_reftime)
+
+    # TODO: Think about temperature units in relation to
+    #       https://github.com/cf-convention/discuss/issues/101,
+    #       whenever that issue is resolved.
```

Comment:
Nice to see we are ahead of the game by having this function ready to slot in the appropriate logic corresponding to the outcome of that Issue (if it ever gets decided upon!) :slightly_smiling_face: